src/prefect/experimental/cli/artifact.py

```diff
@@ -132,3 +136,81 @@ async def inspect(
         artifacts = [a.dict(json_compatible=True) for a in artifacts]
 
         app.console.print(Pretty(artifacts))
+
+
+@artifact_app.command("delete")
+@experimental(
+    feature="The Artifact CLI",
+    group="artifacts",
+)
+async def delete(
+    key: Optional[str] = typer.Argument(
+        None, help="The key of the artifact to delete."
+    ),
+    artifact_id: Optional[str] = typer.Option(
+        None, "--id", help="The ID of the artifact to delete."
+    ),
+):
+    """
+    Delete an artifact.
+
+    Arguments:
+        key: the key of the artifact to delete
+
+    Examples:
+        $ prefect artifact delete "my-artifact"
+    """
+    if key and artifact_id:
+        exit_with_error("Please provide either a key or an artifact_id but not both.")
+
+    async with get_client() as client:
+        if artifact_id is not None:
+            try:
+                confirm_delete = typer.confirm(
+                    (
+                        "Are you sure you want to delete artifact with id"
+                        f" {artifact_id!r}?"
+                    ),
+                    default=False,
+                )
+                if not confirm_delete:
+                    exit_with_error("Deletion aborted.")
+
+                await client.delete_artifact(artifact_id)
+                exit_with_success(f"Deleted artifact with id {artifact_id!r}.")
+            except ObjectNotFound:
+                exit_with_error(f"Artifact with id {artifact_id!r} not found!")
+
+        elif key is not None:
+            BATCH_SIZE = 100
+
+            artifacts = await client.read_artifacts(
+                artifact_filter=schemas.filters.ArtifactFilter(
+                    key=schemas.filters.ArtifactFilterKey(any_=[key])
+                ),
+            )
+            if not artifacts:
+                exit_with_error(f"Artifact {key!r} not found.")
+
+            confirm_delete = typer.confirm(
+                (
+                    f"Are you sure you want to delete {len(artifacts)} artifact(s) with"
+                    f" key {key!r}?"
+                ),
+                default=False,
+            )
+            if not confirm_delete:
+                exit_with_error("Deletion aborted.")
+
+            async def delete_artifacts(artifact_batch):
+                tasks = [client.delete_artifact(str(a.id)) for a in artifact_batch]
+                await asyncio.gather(*tasks)
```

Comment:
Moving this batching delete logic to an API route might be worthwhile. Performing up to 100 API calls concurrently could take a long time since we allow a maximum of 16 open connections at a time.

Reply:
+1

Reply:
Makes sense!